agent/pkg/k8s/k8s.go

```diff
@@ -92,23 +92,24 @@ func NewClient(sharedInformers informers.SharedInformerFactory, k8sClient *l5dk8
 	// For types that we do not have a cleint for we use the `unstructured` package. As we
 	// also do not have proto definitions for these CRDs, we serialize them to JSON.
 	//
-	// +-------------------------+-----------------------------+----------+------------+
-	// |          Group          |            Kind             |  Client  | Serializer |
-	// +-------------------------+-----------------------------+----------+------------+
-	// | policy.linkerd.io       | servers,serverAuthorizaions | dynamic  | json       |
-	// | multicluster.linkerd.io | links                       | dynamic  | json       |
-	// | linkerd.io              | serviceprofiles             | spclient | json       |
-	// | split.smi-spec.io       | trafficsplits               | tsclient | json       |
-	// +-------------------------+-----------------------------+----------+------------+
+	// +-------------------------+---------------------+-----------+------------+
+	// |          Group          |        Kind         |  Client   | Serializer |
+	// +-------------------------+---------------------+-----------+------------+
+	// | policy.linkerd.io       | serverAuthorizaions | l5dClient | json       |
+	// | policy.linkerd.io       | servers             | l5dClient | json       |
+	// | multicluster.linkerd.io | links               | dynamic   | json       |
+	// | linkerd.io              | serviceprofiles     | spclient  | json       |
+	// | split.smi-spec.io       | trafficsplits       | tsclient  | json       |
```

Comment:
We are using the `tsclient` in non test code right here: https://github.com/BuoyantIO/linkerd-buoyant/blob/zd/use-policy-crds/agent/pkg/k8s/traffic_split.go#L21. `tsclient` is not part of  `l5dClient` as `l5dClient` is coming from the codegened clients in the Linkerd repo. `tsclient` is an SMI client that is coming from another repo. It just happens so that it is being used in the `l5dk8s.KubernetesAPI{}` and this is why we access it from there. So we really have two clients that we depend on: 

- `k8sClient *l5dk8s.KubernetesAPI `- this is the Linkerd client that contains the TsClient and the Dynamic client
- `l5dClient l5dApi.Interface`  - the client that is coming from the Linkerd repo that gives us clients for all Linkerd specific CRDs (not that TS is not part of these..)

If you ask me, these can be combined into one in the Linkerd repo and that would make things cleaner but this is a separate task I think.